### PR TITLE
Adds PropertyExpression helper method

### PR DESCRIPTION
**Pull request by cheeezburga**
### Description
This PR adds a `getPatterns` helper method to PropertyExpression.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java [hunk 1]
```diff
@@ -36,43 +18,58 @@
...
- * 
+ *
  * @see SimplePropertyExpression
  * @see #register(Class, Class, String, String)
  */
 public abstract class PropertyExpression<F, T> extends SimpleExpression<T> {
 
+	/**
+	 * A helper method to get the two default property patterns from a property and type parameter.
+	 * i.e. "property of %types%" and "%types%'[s] property"
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java:29]:
```suggestion
	 * i.e. "[the] property of %types%" and "%types%'[s] property"
```

File: src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java [hunk 2]
```diff
@@ -36,43 +18,58 @@
...
+	 *
+	 * @param property the name of property
+	 * @param fromType the types that the property should apply to
+	 * @return an array of strings that represent the patterns of the provided property and types
+	 */
+	@SuppressWarnings("ConstantValue")
+	public static String[] getPatterns(String property, String fromType) {
+		if (property == null || fromType == null)
+			return null;
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java:38]: i would suggest an exception here instead, there's no case in which one of these being null is intended

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java [hunk 3]
```diff
@@ -36,43 +18,58 @@
...
-	 * 
+	 *
 	 * @param expressionClass the PropertyExpression class being registered.
 	 * @param type the main expression type the property is based off of.
 	 * @param property the name of the property.
 	 * @param fromType should be plural to support multiple objects but doesn't have to be.
 	 */
 	public static <T> void registerDefault(Class<? extends Expression<T>> expressionClass, Class<T> type, String property, String fromType) {
-		Skript.registerExpression(expressionClass, type, ExpressionType.PROPERTY, "[the] " + property + " [of %" + fromType + "%]", "%" + fromType + "%'[s] " + property);
+		Skript.registerExpression(expressionClass, type, ExpressionType.PROPERTY, getPatterns(property, fromType));
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java:64]: This is not registering a pattern with a default expression anymore

**Review by UnderscoreTud — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java [hunk 4]
```diff
@@ -30,13 +30,15 @@ public abstract class PropertyExpression<F, T> extends SimpleExpression<T> {
 	 *
 	 * @param property the name of property
 	 * @param fromType the types that the property should apply to
+	 * @param defaultExpression whether the types should be optional
 	 * @return an array of strings that represent the patterns of the provided property and types
 	 */
 	@SuppressWarnings("ConstantValue")
-	public static String[] getPatterns(String property, String fromType) {
+	public static String[] getPatterns(String property, String fromType, boolean defaultExpression) {
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/expressions/base/PropertyExpression.java:37]: You should overload the method and default `defaultExpression` to false.
Or better yet, remove the boolean parameter and make them two separate methods.

**Review by UnderscoreTud — CHANGES_REQUESTED** [review]

**Review by UnderscoreTud — APPROVED** [review]

**Review by Moderocky — APPROVED** [review]
Good.

**Pull request merged**